### DISCUSSION TITLE
Parameter for max translation speed in Goal: Drive Arc

### DISCUSSION THREAD

**Discussion by gruedisueli** [iRobotEducation/create3_docs#425] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 22.04.02LTS on Oracle VM

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

G.4.5

### Which RMW is your robot running?

CycloneDDS

### Does your robot have an assigned namespace? If so, please share.

maeve and dolores

### Is the robot connected to a network? If so, what is the network type?

dedicated network for ros2

### Are there multiple Create 3 robots connected to your network?

Yes

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I would like to know what the units are for the max translation speed parameter in the Goal DriveArc. It's stated in the [documentation ](https://iroboteducation.github.io/create3_docs/examples/actuators-cli/)that this is meters per second, but my observations seem to indicate otherwise. 

I performed an experiment where I to drove two iRobots in concentric circles. The smaller circle had radius R. The outer circle had radius 2(R). The total arc angle was 2(PI) radians. The direction is "Forward"/"1". The max translation speed for the iRobot in the smaller circle was V, the max speed for the outer iRobot was V(2). 

My expectation with the above parameters would be that both robots would start and end the circle maneuver at the same time, or very close to it. 

My observation is that the inner robot completes the circle in <50% the time of the outer one.

Is there  something I'm missing with the parameter setup? Is there some nuance to what the translation speed is referring to that I'm missing?

**Comment by justinIRBT** [comment]
The maximum speed of the robot depends on the safety mode as defined here: https://iroboteducation.github.io/create3_docs/api/safety/ (see "Maximum speed" section).
In default safety mode it is 0.306 m/s.
An example usage is here:
https://iroboteducation.github.io/create3_docs/api/drive-goals/
```
ros2 action send_goal /drive_arc irobot_create_msgs/action/DriveArc "{angle: 1.57,radius: 0.3,translate_direction: 1,max_translation_speed: 0.3}"
```
What are the values you are using?

**Comment by gruedisueli** [comment]
I override all safeties.

The values I used were 0.4 and 0.2 for the outer and inner, respectively.

**Reply by justinIRBT** [reply]
I am wondering if you are saturating the wheels with the combination of rotate and translate, the 0.46 max speed is total wheel speed, so if you are translating at 0.4, that only leave 600mm/s for rotate.  The robot will do the computations for the wheel given the parameters without applying limits first, but if a wheel is above the max velocity, it will scale both wheels down until the larger wheel is at max to try to keep the same arc trajectory, but not at the requested max speed.  If you are saturating wheel velocity at different magnitudes then they would scale differently relative to each other.  Out of curiosity, can you try your experiment with very slow values (one 2x the other) to see if your expectations are met there then scale up the values.
